src/components/FilmForm/FilmForm.tsx

```diff
@@ -31,28 +38,38 @@ const FilmForm: React.FC<FormProps> = ({ film, modalType }) => {
   const searchTitle: string = useSelector<RootState, string>(
     searchedFilmSelector
   );
+  const offset: number = useSelector<RootState, number>(offsetSelector);
+  const sortingType: string = useSelector<RootState, string>(
+    sortingTypeSelector
+  );
 
   const formik = useFormik({
     initialValues: { ...film, genres: stringToObject(film.genres) },
+    validationSchema: SignupSchema,
     onSubmit: (values) => {
+      dispatch(clearfilmsList());
       if (modalType === FormType.EDIT) {
         dispatch(
           formEditFilm.request({
             film: {
               ...values,
               genres: objectToString(values.genres),
             },
-            genre: genre,
-            searchTitle: searchTitle,
+            genre,
+            searchTitle,
+            offset,
+            sortingType,
```

Comment:
object is duplicated in edit and add dispatch, you can prepare this object before `if(modalType === ..`